addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -2043,6 +2043,11 @@ var FieldMany2ManyBinaryMultiFiles = AbstractField.extend({
             })));
         this.$('.oe_fileupload').show();
 
+        // set the attribute ``accept`` on ``.o_input_file`` to filter file extensions
+        if(this.nodeOptions && this.nodeOptions.accepted_file_extensions){
```

Comment:
I think this should be done directly in the template file.

There is apparently already a `image_only` property that you can set on the widget to alter the accept property on the generated input.
Maybe you should change all calls using `image_only` to:

`<t t-set="input_accept" t-value="images/*">` when calling `HiddenInputFile`
Or even better setting your new option on the related widget when possible (have to check case by case).

Then in the base `FieldBinaryFileUploader`, when calling `HiddenInputFile`, add:

`<t t-if="widget.options.accepted_file_extensions" t-set="input_accept" t-value="widget.options.accepted_file_extensions" />`

Then finally in the `HiddenInputFile` template itself, add:

`<input" type="file" class="o_input_file" name="ufile" t-att="{'multiple': multi_upload ? 'multiple' : null}" t-att-accept="input_accept || '*'"/>`

This will get rid of image_only to enable a more flexible approach and add the accept property at the template level instead of the post render phase.

My opinion, but @VincentSchippefilt will have more insights on this.

Reply:
@awa-odoo @VincentSchippefilt maybe it's even more simple to change the input of ``HiddenInputFile`` to
```xml
<input type="file" class="o_input_file" name="ufile"
                t-att="{'multiple': multi_upload ? 'multiple' : null, 'accept': widget.nodeOptions.accepted_file_extensions || '*'}"/>
```

So, all widgets using ``HiddenInputFile`` will be able to use this option (and we have less code, just on line in XML)